### ISSUE TITLE
Karma Server not starting

### ISSUE THREAD

**Issue by asylejmani** [karma-runner/karma-intellij#26]
Hello,

I create the karma.conf.js and it uses the karma plugin from webstorm, but when I try to run it, I get this error all the time:

/usr/local/bin/node /Applications/WebStorm.app/Contents/plugins/js-karma/js_reporter/karma-intellij/lib/intellijServer.js --karmaPackageDir=/usr/local/lib/node_modules/karma --configFile=/Users/arber/Sites/portal/tests/karma.conf.js

/Applications/WebStorm.app/Contents/plugins/js-karma/js_reporter/karma-intellij/lib/intellijServer.js:10
server.start(cliOptions);
       ^
TypeError: Object function (cliOptions, done) {
  EventEmitter.call(this)

  setupLogger(cliOptions.logLevel, cliOptions.colors)

  this.log = logger.create()

  var config = cfg.parseConfig(cliOptions.configFile, cliOptions)

  var modules = [{
    helper: ['value', helper],
    logger: ['value', logger],
    done: ['value', done || process.exit],
    emitter: ['value', this],
    launcher: ['type', Launcher],
    config: ['value', config],
    preprocess: ['factory', preprocessor.createPreprocessor],
    fileList: ['type', FileList],
    webServer: ['factory', ws.create],
    socketServer: ['factory', createSocketIoServer],
    executor: ['type', Executor],
    // TODO(vojta): remove
    customFileHandlers: ['value', []],
    // TODO(vojta): remove, once karma-dart does not rely on it
    customScriptTypes: ['value', []],
    reporter: ['factory', reporter.createReporters],
    capturedBrowsers: ['type', BrowserCollection],
    args: ['value', {}],
    timer: ['value', {setTimeout: setTimeout, clearTimeout: clearTimeout}]
  }]

  // Load the plugins
  modules = modules.concat(plugin.resolve(config.plugins))

  this._injector = new di.Injector(modules)
} has no method 'start'
    at Object.<anonymous> (/Applications/WebStorm.app/Contents/plugins/js-karma/js_reporter/karma-intellij/lib/intellijServer.js:10:8)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:906:3

Process finished with exit code 8

I'm using Webstorm 10 with karma plugin version: 141.1550

Thanks.

**Comment by albuch** [comment]
Similar issue here with IDEA 14.1.4 and Karma Plugin 141.1108.
Karma works on command line. Starting per Karma Run/Debug Configuration stops with 

    "C:\Program Files\nodejs\node.exe" C:\Users\test\.IntelliJIdea14\config\plugins\js-karma\js_reporter\karma-intellij\lib\intellijServer.js --karmaPackageDir=C:\Users\test\AppData\Roaming\npm\node_modules\karma --configFile=D:\test\karma.conf.js
    C:\Users\test\.IntelliJIdea14\config\plugins\js-karma\js_reporter\karma-intellij\lib\intellijServer.js:10
    server.start(cliOptions);
           ^
     TypeError: undefined is not a function
    at Object.<anonymous> (C:\Users\test\.IntelliJIdea14\config\plugins\js-karma\js_reporter\karma-intellij\lib\intellijServer.js:10:8)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
    at node.js:814:3

    Process finished with exit code 1

**Comment by willykc** [comment]
This is introduced by a breaking change in version 0.13.x of Karma. Quick workaround is to change intellijServer.js to this:

```javascript
var cli = require('./intellijCli.js')
  , Server = cli.requireKarmaModule('lib/server.js') // <-- Server is a class now
  , cliOptions = { configFile: require.resolve('./intellij.conf.js') };

var browsers = cli.getBrowsers();
if (browsers != null) {
  cliOptions.browsers = browsers;
}

var server = new Server(cliOptions); // <-- Instantiate the class with cliOptions
server.start(); // <-- Start server without any parameters

// Prevent karma server from being an orphan process.
// For example, if WebStorm is killed using SIGKILL, karma server will still be alive.
// When WebStorm is terminated, karma server's standard input is closed automatically.
process.stdin.resume();
process.stdin.on('close', function () {
  // terminating orphan process
  process.exit(123);
});
```

**Comment by segrey** [comment]
Thanks for the heads up.
The fix will be available in WebStorm 11 (and in WebStorm 10.0.5 if it's released - no such plans yet).
